In [38]:
import pandas as pd
import numpy as np
from datetime import datetime
from datetime import date
import matplotlib.pyplot as plt
import requests
pd.options.display.max_rows = 5000 
pd.options.display.max_columns = 200

cobr_p1

In [67]:
cobr = pd.read_csv(r'/Users/paxton615/Desktop/Stock_Price_csv/cobr-2021-10.csv')
cobr.drop('Unnamed: 0',axis=1, inplace=True)

# ----------------------------------------------------------

p1_date = '2020-01-01'
p1_price = 1.00
p1_wprice = 2.00

twenty_per_up = p1_price * (1+0.2)
thirty_per_up = p1_price * (1+0.3)
fourty_per_up = p1_price * (1+0.4)
fifty_per_up = p1_price * (1+0.5)

cobr_p1 = cobr[cobr['cobr_dates']>=p1_date]
cobr_p1 = cobr_p1[::-1].reset_index(drop=True)

# ----------------------------------------------------------

trade_high = cobr_p1['cobr_close'].max()
trade_high_date = (cobr_p1.loc[cobr_p1['cobr_close']==cobr_p1['cobr_close'].max(), 'cobr_dates'].item())
highest_rtn = (cobr_p1['cobr_close'].max()- p1_price + cobr_p1['cobr_close'].max() - p1_wprice)/p1_price


cobr_p1_df = pd.DataFrame({'placing':'cobr_p1', 'placing_date':p1_date, 'placing_price':p1_price, 'warrant_price':p1_wprice ,
                          'trade_high':trade_high, 'trade_high_date':trade_high_date, 'highest_rtn':highest_rtn},index=[0])


cobr_p1_df['wprice_bar'] = [0 if ((p1_wprice-p1_price)/p1_price)<=0 else (p1_wprice-p1_price)/p1_price ][0]
cobr_p1_df['over_w_days'] = [0 if p1_wprice==0 else len(cobr_p1[cobr_p1['cobr_close']>= p1_wprice])][0]

cobr_p1_df['all_trade_days'] = len(cobr_p1.cobr_close)
cobr_p1_df['20%up_days'] = len(cobr_p1[cobr_p1['cobr_close']>= twenty_per_up].cobr_close)
cobr_p1_df['30%up_days'] = len(cobr_p1[cobr_p1['cobr_close']>= thirty_per_up].cobr_close)
cobr_p1_df['40%up_days'] = len(cobr_p1[cobr_p1['cobr_close']>= fourty_per_up].cobr_close)
cobr_p1_df['50%up_days'] = len(cobr_p1[cobr_p1['cobr_close']>= fifty_per_up].cobr_close)
cobr_p1_df['200%up_days'] = len(cobr_p1[cobr_p1['cobr_close']>= p1_price*2].cobr_close)
cobr_p1_df['300%up_days'] = len(cobr_p1[cobr_p1['cobr_close']>= p1_price*3].cobr_close)
cobr_p1_df['400%up_days'] = len(cobr_p1[cobr_p1['cobr_close']>= p1_price*4].cobr_close)
cobr_p1_df['500%up_days'] = len(cobr_p1[cobr_p1['cobr_close']>= p1_price*5].cobr_close)
cobr_p1_df['600%up_days'] = len(cobr_p1[cobr_p1['cobr_close']>= p1_price*6].cobr_close)
cobr_p1_df['700%up_days'] = len(cobr_p1[cobr_p1['cobr_close']>= p1_price*7].cobr_close)
cobr_p1_df['800%up_days'] = len(cobr_p1[cobr_p1['cobr_close']>= p1_price*8].cobr_close)
cobr_p1_df['900%up_days'] = len(cobr_p1[cobr_p1['cobr_close']>= p1_price*9].cobr_close)

def first_profit_day_number(column, target_price):
    count = 0
    for i in column:
        if i < target_price:
            count += 1
        if i > target_price:
            break
    return count

cobr_p1_df['0_to_0%+'] = first_profit_day_number(cobr_p1['cobr_close'], p1_price)
cobr_p1_df['0_to_20%+'] = first_profit_day_number(cobr_p1['cobr_close'], twenty_per_up)
cobr_p1_df['0_to_30%+'] = first_profit_day_number(cobr_p1['cobr_close'], thirty_per_up)
cobr_p1_df['0_to_40%+'] = first_profit_day_number(cobr_p1['cobr_close'], fourty_per_up)
cobr_p1_df['0_to_50%+'] = first_profit_day_number(cobr_p1['cobr_close'], fifty_per_up)

cobr_p1_df

placing placing_date  placing_price  warrant_price  trade_high  \
0  cobr_p1   2020-01-01            1.0            2.0         3.5   

  trade_high_date  highest_rtn  wprice_bar  over_w_days  all_trade_days  \
0      2020-10-20          4.0         1.0          225             442   

   20%up_days  30%up_days  40%up_days  50%up_days  200%up_days  300%up_days  \
0         408         394         387         383          225           24   

   400%up_days  500%up_days  600%up_days  700%up_days  800%up_days  \
0            0            0            0            0            0   

   900%up_days  0_to_0%+  0_to_20%+  0_to_30%+  0_to_40%+  0_to_50%+  
0            0         0          0          0         22         22

cobr_p1 , cobr_p2

In [65]:
cobr = pd.read_csv(r'/Users/paxton615/Desktop/Stock_Price_csv/cobr-2021-10.csv')
cobr.drop('Unnamed: 0',axis=1, inplace=True)

# ----------------------------------------------------------

p2_date = '2020-09-01'
p2_price = 2.30
p2_wprice = 3.00

twenty_per_up = p2_price * (1+0.2)
thirty_per_up = p2_price * (1+0.3)
fourty_per_up = p2_price * (1+0.4)
fifty_per_up = p2_price * (1+0.5)

cobr_p2 = cobr[cobr['cobr_dates']>=p2_date]
cobr_p2 = cobr_p2[::-1].reset_index(drop=True)

# ----------------------------------------------------------

trade_high = cobr_p2['cobr_close'].max()
trade_high_date = (cobr_p2.loc[cobr_p2['cobr_close']==cobr_p2['cobr_close'].max(), 'cobr_dates'].item())
highest_rtn = (cobr_p2['cobr_close'].max()- p2_price + cobr_p2['cobr_close'].max() - p2_wprice)/p2_price


cobr_p2_df = pd.DataFrame({'placing':'cobr_p2', 'placing_date':p2_date, 'placing_price':p2_price, 'warrant_price':p2_wprice ,
                          'trade_high':trade_high, 'trade_high_date':trade_high_date, 'highest_rtn':highest_rtn},index=[0])


cobr_p2_df['wprice_bar'] = [0 if ((p2_wprice-p2_price)/p2_price)<=0 else (p2_wprice-p2_price)/p2_price ][0]
cobr_p2_df['over_w_days'] = [0 if p2_wprice==0 else len(cobr_p2[cobr_p2['cobr_close']>= p2_wprice])][0]

cobr_p2_df['all_trade_days'] = len(cobr_p2.cobr_close)
cobr_p2_df['20%up_days'] = len(cobr_p2[cobr_p2['cobr_close']>= twenty_per_up].cobr_close)
cobr_p2_df['30%up_days'] = len(cobr_p2[cobr_p2['cobr_close']>= thirty_per_up].cobr_close)
cobr_p2_df['40%up_days'] = len(cobr_p2[cobr_p2['cobr_close']>= fourty_per_up].cobr_close)
cobr_p2_df['50%up_days'] = len(cobr_p2[cobr_p2['cobr_close']>= fifty_per_up].cobr_close)
cobr_p2_df['200%up_days'] = len(cobr_p2[cobr_p2['cobr_close']>= p2_price*2].cobr_close)
cobr_p2_df['300%up_days'] = len(cobr_p2[cobr_p2['cobr_close']>= p2_price*3].cobr_close)
cobr_p2_df['400%up_days'] = len(cobr_p2[cobr_p2['cobr_close']>= p2_price*4].cobr_close)
cobr_p2_df['500%up_days'] = len(cobr_p2[cobr_p2['cobr_close']>= p2_price*5].cobr_close)
cobr_p2_df['600%up_days'] = len(cobr_p2[cobr_p2['cobr_close']>= p2_price*6].cobr_close)
cobr_p2_df['700%up_days'] = len(cobr_p2[cobr_p2['cobr_close']>= p2_price*7].cobr_close)
cobr_p2_df['800%up_days'] = len(cobr_p2[cobr_p2['cobr_close']>= p2_price*8].cobr_close)
cobr_p2_df['900%up_days'] = len(cobr_p2[cobr_p2['cobr_close']>= p2_price*9].cobr_close)



def first_profit_day_number(column, target_price):
    count = 0
    for i in column:
        if i < target_price:
            count += 1
        if i > target_price:
            break
    return count

cobr_p2_df['0_to_0%+'] = first_profit_day_number(cobr_p2['cobr_close'], p2_price)
cobr_p2_df['0_to_20%+'] = first_profit_day_number(cobr_p2['cobr_close'], twenty_per_up)
cobr_p2_df['0_to_30%+'] = first_profit_day_number(cobr_p2['cobr_close'], thirty_per_up)
cobr_p2_df['0_to_40%+'] = first_profit_day_number(cobr_p2['cobr_close'], fourty_per_up)
cobr_p2_df['0_to_50%+'] = first_profit_day_number(cobr_p2['cobr_close'], fifty_per_up)


cobr_p2_df

placing placing_date  placing_price  warrant_price  trade_high  \
0  cobr_p2   2020-09-01            2.3            3.0         3.5   

  trade_high_date  highest_rtn  wprice_bar  over_w_days  all_trade_days  \
0      2020-10-20      0.73913    0.304348           24             284   

   20%up_days  30%up_days  40%up_days  50%up_days  200%up_days  300%up_days  \
0          43          24           7           1            0            0   

   400%up_days  500%up_days  600%up_days  700%up_days  800%up_days  \
0            0            0            0            0            0   

   900%up_days  0_to_0%+  0_to_20%+  0_to_30%+  0_to_40%+  0_to_50%+  
0            0         8         34         34         35         35

### cobr_p1 , cobr_p2, pre_p1

In [64]:
pre = pd.read_csv(r'/Users/paxton615/Desktop/Stock_Price_csv/pre-2021-10.csv')
pre.drop('Unnamed: 0',axis=1, inplace=True)

# ----------------------------------------------------------

p1_date = '2021-06-01'
p1_price = 120.00
p1_wprice = 0.00

twenty_per_up = p1_price * (1+0.2)
thirty_per_up = p1_price * (1+0.3)
fourty_per_up = p1_price * (1+0.4)
fifty_per_up = p1_price * (1+0.5)

pre_p1 = pre[pre['pre_dates']>=p1_date]
pre_p1 = pre_p1[::-1].reset_index(drop=True)

# ----------------------------------------------------------

trade_high = pre_p1['pre_close'].max()
trade_high_date = (pre_p1.loc[pre_p1['pre_close']==pre_p1['pre_close'].max(), 'pre_dates'].item())
highest_rtn = (pre_p1['pre_close'].max()- p1_price + pre_p1['pre_close'].max() - p1_wprice)/p1_price


pre_p1_df = pd.DataFrame({'placing':'pre_p1', 'placing_date':p1_date, 'placing_price':p1_price, 'warrant_price':p1_wprice ,
                          'trade_high':trade_high, 'trade_high_date':trade_high_date, 'highest_rtn':highest_rtn},index=[0])


pre_p1_df['wprice_bar'] = [0 if ((p1_wprice-p1_price)/p1_price)<=0 else (p1_wprice-p1_price)/p1_price ][0]
pre_p1_df['over_w_days'] = [0 if p1_wprice==0 else len(pre_p1[pre_p1['pre_close']>= p1_wprice])][0]

pre_p1_df['all_trade_days'] = len(pre_p1.pre_close)
pre_p1_df['20%up_days'] = len(pre_p1[pre_p1['pre_close']>= twenty_per_up].pre_close)
pre_p1_df['30%up_days'] = len(pre_p1[pre_p1['pre_close']>= thirty_per_up].pre_close)
pre_p1_df['40%up_days'] = len(pre_p1[pre_p1['pre_close']>= fourty_per_up].pre_close)
pre_p1_df['50%up_days'] = len(pre_p1[pre_p1['pre_close']>= fifty_per_up].pre_close)
pre_p1_df['200%up_days'] = len(pre_p1[pre_p1['pre_close']>= p1_price*2].pre_close)
pre_p1_df['300%up_days'] = len(pre_p1[pre_p1['pre_close']>= p1_price*3].pre_close)
pre_p1_df['400%up_days'] = len(pre_p1[pre_p1['pre_close']>= p1_price*4].pre_close)
pre_p1_df['500%up_days'] = len(pre_p1[pre_p1['pre_close']>= p1_price*5].pre_close)
pre_p1_df['600%up_days'] = len(pre_p1[pre_p1['pre_close']>= p1_price*6].pre_close)
pre_p1_df['700%up_days'] = len(pre_p1[pre_p1['pre_close']>= p1_price*7].pre_close)
pre_p1_df['800%up_days'] = len(pre_p1[pre_p1['pre_close']>= p1_price*8].pre_close)
pre_p1_df['900%up_days'] = len(pre_p1[pre_p1['pre_close']>= p1_price*9].pre_close)


def first_profit_day_number(column, target_price):
    count = 0
    for i in column:
        if i < target_price:
            count += 1
        if i > target_price:
            break
    return count

pre_p1_df['0_to_0%+'] = first_profit_day_number(pre_p1['pre_close'], p1_price)
pre_p1_df['0_to_20%+'] = first_profit_day_number(pre_p1['pre_close'], twenty_per_up)
pre_p1_df['0_to_30%+'] = first_profit_day_number(pre_p1['pre_close'], thirty_per_up)
pre_p1_df['0_to_40%+'] = first_profit_day_number(pre_p1['pre_close'], fourty_per_up)
pre_p1_df['0_to_50%+'] = first_profit_day_number(pre_p1['pre_close'], fifty_per_up)

pre_p1_df

placing placing_date  placing_price  warrant_price  trade_high  \
0  pre_p1   2021-06-01          120.0            0.0       129.0   

  trade_high_date  highest_rtn  wprice_bar  over_w_days  all_trade_days  \
0      2021-06-11         1.15           0            0              96   

   20%up_days  30%up_days  40%up_days  50%up_days  200%up_days  300%up_days  \
0           0           0           0           0            0            0   

   400%up_days  500%up_days  600%up_days  700%up_days  800%up_days  \
0            0            0            0            0            0   

   900%up_days  0_to_0%+  0_to_20%+  0_to_30%+  0_to_40%+  0_to_50%+  
0            0         7         96         96         96         96

In [36]:
frames = [cobr_p1_df,cobr_p2_df, pre_p1_df]

In [41]:
pd.concat(frames)

placing placing_date  placing_price  warrant_price  trade_high  \
0  cobr_p1   2020-01-01            1.0            2.0         3.5   
0  cobr_p2   2020-09-01            2.3            3.0         3.5   
0   pre_p1   2021-06-01          120.0            0.0       129.0   

  trade_high_date  highest_rtn  wprice_bar  over_w_days  all_trade_days  \
0      2020-10-20      4.00000    1.000000          225             442   
0      2020-10-20      0.73913    0.304348           24             284   
0      2021-06-11      1.15000    0.000000            0              96   

   20%up_days  30%up_days  40%up_days  50%up_days  200%up_days  300%up_days  \
0         408         394         387         383          225           24   
0          43          24           7           1            0            0   
0           0           0           0           0            0            0   

   400%up_days  500%up_days  600%up_days  700%up_days  800%up_days  \
0            0            0            0            0            0   
0            0            0            0            0            0   
0            0            0            0            0            0   

   900%up_days  0_to_0%+  0_to_20%+  0_to_30%+  0_to_40%+  0_to_50%+  
0            0         0          0          0         22         22  
0            0         8         34         34         35         35  
0            0         7         96         96         96         96

### cobr_p1 , cobr_p2, pre_p1, 

In [68]:
pre = pd.read_csv(r'/Users/paxton615/Desktop/Stock_Price_csv/pre-2021-10.csv')
pre.drop('Unnamed: 0',axis=1, inplace=True)

# ----------------------------------------------------------

p1_date = '2021-06-01'
p1_price = 120.00
p1_wprice = 0.00

twenty_per_up = p1_price * (1+0.2)
thirty_per_up = p1_price * (1+0.3)
fourty_per_up = p1_price * (1+0.4)
fifty_per_up = p1_price * (1+0.5)

pre_p1 = pre[pre['pre_dates']>=p1_date]
pre_p1 = pre_p1[::-1].reset_index(drop=True)

# ----------------------------------------------------------

trade_high = pre_p1['pre_close'].max()
trade_high_date = (pre_p1.loc[pre_p1['pre_close']==pre_p1['pre_close'].max(), 'pre_dates'].item())
highest_rtn = (pre_p1['pre_close'].max()- p1_price + pre_p1['pre_close'].max() - p1_wprice)/p1_price


pre_p1_df = pd.DataFrame({'placing':'pre_p1', 'placing_date':p1_date, 'placing_price':p1_price, 'warrant_price':p1_wprice ,
                          'trade_high':trade_high, 'trade_high_date':trade_high_date, 'highest_rtn':highest_rtn},index=[0])


pre_p1_df['wprice_bar'] = [0 if ((p1_wprice-p1_price)/p1_price)<=0 else (p1_wprice-p1_price)/p1_price ][0]
pre_p1_df['over_w_days'] = [0 if p1_wprice==0 else len(pre_p1[pre_p1['pre_close']>= p1_wprice])][0]

pre_p1_df['all_trade_days'] = len(pre_p1.pre_close)
pre_p1_df['20%up_days'] = len(pre_p1[pre_p1['pre_close']>= twenty_per_up].pre_close)
pre_p1_df['30%up_days'] = len(pre_p1[pre_p1['pre_close']>= thirty_per_up].pre_close)
pre_p1_df['40%up_days'] = len(pre_p1[pre_p1['pre_close']>= fourty_per_up].pre_close)
pre_p1_df['50%up_days'] = len(pre_p1[pre_p1['pre_close']>= fifty_per_up].pre_close)
pre_p1_df['200%up_days'] = len(pre_p1[pre_p1['pre_close']>= p1_price*2].pre_close)
pre_p1_df['300%up_days'] = len(pre_p1[pre_p1['pre_close']>= p1_price*3].pre_close)
pre_p1_df['400%up_days'] = len(pre_p1[pre_p1['pre_close']>= p1_price*4].pre_close)
pre_p1_df['500%up_days'] = len(pre_p1[pre_p1['pre_close']>= p1_price*5].pre_close)
pre_p1_df['600%up_days'] = len(pre_p1[pre_p1['pre_close']>= p1_price*6].pre_close)
pre_p1_df['700%up_days'] = len(pre_p1[pre_p1['pre_close']>= p1_price*7].pre_close)
pre_p1_df['800%up_days'] = len(pre_p1[pre_p1['pre_close']>= p1_price*8].pre_close)
pre_p1_df['900%up_days'] = len(pre_p1[pre_p1['pre_close']>= p1_price*9].pre_close)


def first_profit_day_number(column, target_price):
    count = 0
    for i in column:
        if i < target_price:
            count += 1
        if i > target_price:
            break
    return count

pre_p1_df['0_to_0%+'] = first_profit_day_number(pre_p1['pre_close'], p1_price)
pre_p1_df['0_to_20%+'] = first_profit_day_number(pre_p1['pre_close'], twenty_per_up)
pre_p1_df['0_to_30%+'] = first_profit_day_number(pre_p1['pre_close'], thirty_per_up)
pre_p1_df['0_to_40%+'] = first_profit_day_number(pre_p1['pre_close'], fourty_per_up)
pre_p1_df['0_to_50%+'] = first_profit_day_number(pre_p1['pre_close'], fifty_per_up)

pre_p1_df

placing placing_date  placing_price  warrant_price  trade_high  \
0  pre_p1   2021-06-01          120.0            0.0       129.0   

  trade_high_date  highest_rtn  wprice_bar  over_w_days  all_trade_days  \
0      2021-06-11         1.15           0            0              96   

   20%up_days  30%up_days  40%up_days  50%up_days  200%up_days  300%up_days  \
0           0           0           0           0            0            0   

   400%up_days  500%up_days  600%up_days  700%up_days  800%up_days  \
0            0            0            0            0            0   

   900%up_days  0_to_0%+  0_to_20%+  0_to_30%+  0_to_40%+  0_to_50%+  
0            0         7         96         96         96         96

### acp_p1

In [69]:
acp = pd.read_csv(r'/Users/paxton615/Desktop/Stock_Price_csv/acp-2021-10.csv')
acp.drop('Unnamed: 0',axis=1, inplace=True)

# ----------------------------------------------------------

p1_date = '2019-09-01'
p1_price = 1.80
p1_wprice = 3.00

twenty_per_up = p1_price * (1+0.2)

thirty_per_up = p1_price * (1+0.3)
fourty_per_up = p1_price * (1+0.4)
fifty_per_up = p1_price * (1+0.5)

acp_p1 = acp[acp['acp_dates']>=p1_date]
acp_p1 = acp_p1[::-1].reset_index(drop=True)

# ----------------------------------------------------------

trade_high = acp_p1['acp_close'].max()
trade_high_date = (acp_p1.loc[acp_p1['acp_close']==acp_p1['acp_close'].max(), 'acp_dates'].item())
highest_rtn = (acp_p1['acp_close'].max()- p1_price + acp_p1['acp_close'].max() - p1_wprice)/p1_price


acp_p1_df = pd.DataFrame({'placing':'acp_p1', 'placing_date':p1_date, 'placing_price':p1_price, 'warrant_price':p1_wprice ,
                          'trade_high':trade_high, 'trade_high_date':trade_high_date, 'highest_rtn':highest_rtn},index=[0])


acp_p1_df['wprice_bar'] = [0 if ((p1_wprice-p1_price)/p1_price)<=0 else (p1_wprice-p1_price)/p1_price ][0]
acp_p1_df['over_w_days'] = [0 if p1_wprice==0 else len(acp_p1[acp_p1['acp_close']>= p1_wprice])][0]

acp_p1_df['all_trade_days'] = len(acp_p1.acp_close)
acp_p1_df['20%up_days'] = len(acp_p1[acp_p1['acp_close']>= twenty_per_up].acp_close)
acp_p1_df['30%up_days'] = len(acp_p1[acp_p1['acp_close']>= thirty_per_up].acp_close)
acp_p1_df['40%up_days'] = len(acp_p1[acp_p1['acp_close']>= fourty_per_up].acp_close)
acp_p1_df['50%up_days'] = len(acp_p1[acp_p1['acp_close']>= fifty_per_up].acp_close)
acp_p1_df['200%up_days'] = len(acp_p1[acp_p1['acp_close']>= p1_price*2].acp_close)
acp_p1_df['300%up_days'] = len(acp_p1[acp_p1['acp_close']>= p1_price*3].acp_close)
acp_p1_df['400%up_days'] = len(acp_p1[acp_p1['acp_close']>= p1_price*4].acp_close)
acp_p1_df['500%up_days'] = len(acp_p1[acp_p1['acp_close']>= p1_price*5].acp_close)
acp_p1_df['600%up_days'] = len(acp_p1[acp_p1['acp_close']>= p1_price*6].acp_close)
acp_p1_df['700%up_days'] = len(acp_p1[acp_p1['acp_close']>= p1_price*7].acp_close)
acp_p1_df['800%up_days'] = len(acp_p1[acp_p1['acp_close']>= p1_price*8].acp_close)
acp_p1_df['900%up_days'] = len(acp_p1[acp_p1['acp_close']>= p1_price*9].acp_close)


def first_profit_day_number(column, target_price):
    count = 0
    for i in column:
        if i < target_price:
            count += 1
        if i > target_price:
            break
    return count

acp_p1_df['0_to_0%+'] = first_profit_day_number(acp_p1['acp_close'], p1_price)
acp_p1_df['0_to_20%+'] = first_profit_day_number(acp_p1['acp_close'], twenty_per_up)
acp_p1_df['0_to_30%+'] = first_profit_day_number(acp_p1['acp_close'], thirty_per_up)
acp_p1_df['0_to_40%+'] = first_profit_day_number(acp_p1['acp_close'], fourty_per_up)
acp_p1_df['0_to_50%+'] = first_profit_day_number(acp_p1['acp_close'], fifty_per_up)

acp_p1_df

placing placing_date  placing_price  warrant_price  trade_high  \
0  acp_p1   2019-09-01            1.8            3.0        6.25   

  trade_high_date  highest_rtn  wprice_bar  over_w_days  all_trade_days  \
0      2021-09-14     4.277778    0.666667          417             537   

   20%up_days  30%up_days  40%up_days  50%up_days  200%up_days  300%up_days  \
0         494         470         460         447          264           18   

   400%up_days  500%up_days  600%up_days  700%up_days  800%up_days  \
0            0            0            0            0            0   

   900%up_days  0_to_0%+  0_to_20%+  0_to_30%+  0_to_40%+  0_to_50%+  
0            0         0         13         39         40         44

### 400%up_days 发现问题，428%应该有几天，而我的计算公式是p1的4倍是7.2，比最高6.25的价格高出很多

In [ ]:
acp_p1[(acp_p1['acp_close']>= p1_price*3)&(p1_price*4 > acp_p1['acp_close'])]

In [72]:
acp_p1[(acp_p1['acp_close']>= p1_price*3)&(p1_price*4 > acp_p1['acp_close'])]

acp_dates  acp_close  acp_volume
508  2021-09-03       5.45  46234114.0
509  2021-09-06       5.95  21388896.0
510  2021-09-07       5.85   6922770.0
511  2021-09-08       5.65   3096250.0
512  2021-09-09       6.00  11143348.0
513  2021-09-10       6.05   8662459.0
514  2021-09-13       6.20   1522225.0
515  2021-09-14       6.25   3541856.0
516  2021-09-15       6.05   5930062.0
517  2021-09-16       5.80   5569342.0
518  2021-09-17       5.80   4359798.0
519  2021-09-20       5.70   4891858.0
520  2021-09-21       5.75   3996846.0
521  2021-09-22       5.50   2178289.0
522  2021-09-23       5.50   3571002.0
523  2021-09-24       5.40   5123918.0
524  2021-09-27       5.45   4422712.0
525  2021-09-28       5.50   4897636.0

In [73]:
acp_p1[(acp_p1['acp_close']>= p1_price*4)&(p1_price*5 > acp_p1['acp_close'])]

Empty DataFrame
Columns: [acp_dates, acp_close, acp_volume]
Index: []

In [91]:
aaa = pd.read_csv(r'/Users/paxton615/Desktop/Stock_Price_csv/acp-2021-10.csv')
aaa.drop('Unnamed: 0',axis=1, inplace=True)

# ----------------------------------------------------------

p1_date = '2019-09-01'
p1_price = 1.80
p1_wprice =0.00

twenty_per_up = p1_price * (1+0.2)
thirty_per_up = p1_price * (1+0.3)
fourty_per_up = p1_price * (1+0.4)
fifty_per_up = p1_price * (1+0.5)

aaa_p1 = aaa[aaa['acp_dates']>=p1_date]
aaa_p1 = aaa_p1[::-1].reset_index(drop=True)

# ----------------------------------------------------------

trade_high = aaa_p1['acp_close'].max()
trade_high_date = (aaa_p1.loc[aaa_p1['acp_close']==aaa_p1['acp_close'].max(), 'acp_dates'].item())
highest_rtn = (aaa_p1['acp_close'].max()- p1_price + aaa_p1['acp_close'].max() - p1_wprice)/p1_price

In [92]:
highest_rtn

5.944444444444444

In [93]:
highest_rtn

5.944444444444444

In [95]:
[(aaa_p1['acp_close'].max()- p1_price + aaa_p1['acp_close'].max() - p1_wprice)/p1_price if p1_wprice != 0 else (aaa_p1['acp_close'].max()- p1_price)/p1_price ][0]

4.944444444444445

In [ ]:
[(acp_p1['acp_close'].max()- p1_price + acp_p1['acp_close'].max() - p1_wprice)/p1_price if p1_wprice != 0 else (acp_p1['acp_close'].max()- p1_price)/p1_price ][0]

In [98]:
[(i- p1_price + i - p1_wprice)/p1_price if p1_wprice != 0 else (i- p1_price)/p1_price for i in aaa_p1['acp_close']]

[0.05555555555555548,
 0.05555555555555548,
 0.0277777777777778,
 0.13888888888888876,
 0.11111111111111108,
 0.06944444444444445,
 -0.013888888888888963,
 -0.0277777777777778,
 -0.09722222222222224,
 -0.06944444444444445,
 0.05555555555555548,
 0.08333333333333329,
 0.16666666666666669,
 0.2222222222222223,
 0.2222222222222223,
 0.2777777777777776,
 0.2222222222222223,
 0.2222222222222223,
 0.2777777777777776,
 0.2222222222222223,
 0.16666666666666669,
 0.13888888888888876,
 0.13888888888888876,
 0.11111111111111108,
 0.11111111111111108,
 0.2777777777777776,
 0.2777777777777776,
 0.2716666666666667,
 0.2222222222222223,
 0.2777777777777776,
 0.16666666666666669,
 0.2222222222222223,
 0.16666666666666669,
 0.11111111111111108,
 0.16666666666666669,
 0.2222222222222223,
 0.16666666666666669,
 0.2222222222222223,
 0.2222222222222223,
 0.38888888888888884,
 0.5000000000000001,
 0.4444444444444445,
 0.5000000000000001,
 0.5000000000000001,
 0.47222222222222215,
 0.4444444444444445,
 0.444